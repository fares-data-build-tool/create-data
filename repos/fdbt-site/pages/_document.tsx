import '../design/Pages.scss';
import Document, { Html, Head, Main, NextScript } from 'next/document'
import * as React from 'react';
class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html lang="en" className="govuk-template app-html-class flexbox no-flexboxtweener">
        <Head />
        <body className="govuk-template__body app-body-class">
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument