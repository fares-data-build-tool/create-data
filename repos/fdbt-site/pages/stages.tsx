import '../design/Pages.scss';
import React from 'react';
import { NextPageContext } from 'next';
import Layout from '../layout/Layout';
import { parseCookies } from 'nookies';
import { OPERATOR_COOKIE, SERVICE_COOKIE } from '../constants';
import { getHost, isSessionValid } from '../utils';
import TableForm from '../components/TableForm';


const title = 'Confirmation - Fares data build tool';
const description = 'Confirmation page of the Fares data build tool';

export interface ServiceProps {
  operator: string,
  service: string
}

const Stages = (props: ServiceProps) => {
    return (
      <Layout title={title} description={description}>
          <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
          <p className="govuk-body-l">Welcome operator {props.operator}</p>
            <div className="govuk-form-group">
              <label className="govuk-label" htmlFor="sort">
                Please submit your stages for your service {props.service}
              </label>
            </div>
            <TableForm offset={1} />
          </main>
      </Layout>
    );
}

Stages.getInitialProps = async (ctx: NextPageContext) => {
  function redirectOnError() {
    if (ctx.res) {
      ctx.res.writeHead(302, {
        Location: '/error'
      })
      ctx.res.end();
    }  
  };

  const cookies = parseCookies(ctx);
  const operatorCookie = cookies[OPERATOR_COOKIE];
  const serviceCookie = cookies[SERVICE_COOKIE];
  if(operatorCookie && serviceCookie) {
    const url: string = getHost(ctx.req) + "/api/validate";
    const isValid = await isSessionValid(url, ctx.req);
    if (isValid){
      const operatorObject = JSON.parse(operatorCookie);
      const serviceObject = JSON.parse(serviceCookie);
      return {operator: operatorObject.operator, service: serviceObject.service };
    } else {
      redirectOnError();
    }
  } else {
    redirectOnError();
  }
};

export default Stages;