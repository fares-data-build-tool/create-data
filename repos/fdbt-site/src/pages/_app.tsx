import '../design/Pages.scss';
import '../design/Layout.scss';
import '../design/Components.scss';
import App, { AppProps, AppContext } from 'next/app';
import React, { ReactElement } from 'react';
import { Response } from 'express';
import { CustomAppProps } from '../interfaces';
import logger from '../utils/logger';

const MyApp = ({ Component, pageProps, csrfToken }: AppProps & { csrfToken: string }): ReactElement => (
    // Prop spreading is the recommended way to implement this, from Next.js docs
    // eslint-disable-next-line react/jsx-props-no-spreading
    <Component {...pageProps} csrfToken={csrfToken} />
);

// Force every page to server side render so that we get a chance to set security-related cookies
// This will have a slight performance impact, but it should be negligable
//
// (implementation below from https://nextjs.org/docs/advanced-features/custom-app and
// https://nextjs.org/docs/basic-features/typescript)
//
MyApp.getInitialProps = async (appContext: AppContext): Promise<{}> => {
    let appProps;
    try {
        appProps = await App.getInitialProps(appContext);

        if (appContext.ctx.req && appContext.ctx.res) {
            const { locals } = appContext.ctx.res as Response;
            (appProps as CustomAppProps).csrfToken = locals.csrfToken;
        }
    } catch (error) {
        logger.error(error, { context: 'pages.app', message: 'unhandled exception' });
        throw error;
    }

    // Prop spreading is the recommended way to implement this, from Next.js docs
    // eslint-disable-next-line react/jsx-props-no-spreading
    return { ...appProps };
};

export default MyApp;
