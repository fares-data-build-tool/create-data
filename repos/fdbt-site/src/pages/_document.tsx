import Document, { Html, Head, Main, NextScript, DocumentContext, DocumentInitialProps } from 'next/document';
import React, { ReactElement } from 'react';
import { ServerResponse } from 'http';
import { parseCookies } from 'nookies';
import Header from '../layout/Header';
import Breadcrumbs from '../components/Breadcrumbs';
import { ID_TOKEN_COOKIE } from '../constants';
import AlphaBanner from '../layout/AlphaBanner';
import { Breadcrumb } from '../interfaces';
import Footer from '../layout/Footer';
import breadcrumb from '../utils/breadcrumbs';

interface DocumentProps extends DocumentInitialProps {
    nonce: string;
    isAuthed: boolean;
    csrfToken: string;
    breadcrumbs: Breadcrumb[];
}

interface ResponseWithLocals extends ServerResponse {
    locals: {
        nonce: string;
        csrfToken: string;
    };
}

class MyDocument extends Document<DocumentProps> {
    static async getInitialProps(ctx: DocumentContext): Promise<DocumentProps> {
        const initialProps = await Document.getInitialProps(ctx);
        const nonce = (ctx.res as ResponseWithLocals)?.locals?.nonce ?? null;
        const csrfToken = (ctx.res as ResponseWithLocals)?.locals?.csrfToken ?? null;

        if (ctx.pathname !== '/') {
            ctx.res?.setHeader('X-Robots-Tag', 'none, noindex, nofollow, noimageindex, noarchive');
        }

        const cookies = parseCookies(ctx);
        const idTokenCookie = cookies[ID_TOKEN_COOKIE];

        const breadcrumbs = breadcrumb(ctx).generate();

        return { ...initialProps, nonce, isAuthed: !!idTokenCookie, csrfToken, breadcrumbs };
    }

    render(): ReactElement {
        return (
            <Html lang="en" className="govuk-template app-html-class flexbox no-flexboxtweener">
                <Head nonce={this.props.nonce}>
                    {process.env.STAGE === 'prod' && (
                        <>
                            <script
                                async
                                src="https://www.googletagmanager.com/gtag/js?id=UA-173062045-1"
                                nonce={this.props.nonce}
                            />
                            <script
                                nonce={this.props.nonce}
                                dangerouslySetInnerHTML={{
                                    __html: `window.dataLayer = window.dataLayer || [];
                                        function gtag(){dataLayer.push(arguments);}
                                        gtag('js', new Date());
                                        gtag('config', 'UA-173062045-1');`,
                                }}
                            />
                        </>
                    )}
                </Head>
                <body className="govuk-template__body app-body-class js-enabled">
                    <Header isAuthed={this.props.isAuthed} csrfToken={this.props.csrfToken} />
                    <div className="govuk-width-container app-width-container--wide">
                        <AlphaBanner />
                        {this.props.breadcrumbs.length !== 0 && <Breadcrumbs breadcrumbs={this.props.breadcrumbs} />}
                        <Main />
                    </div>
                    <Footer />

                    <NextScript nonce={this.props.nonce} />
                    <script src="/scripts/all.js" nonce={this.props.nonce} />
                    <script nonce={this.props.nonce}>window.GOVUKFrontend.initAll()</script>
                </body>
            </Html>
        );
    }
}

export default MyDocument;
