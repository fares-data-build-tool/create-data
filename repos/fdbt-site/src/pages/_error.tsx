import React, { ReactElement } from 'react';
import { NextPageContext } from 'next';
import TwoThirdsLayout from '../layout/Layout';
import { FEEDBACK_LINK } from '../constants';

const title = 'Error - Create Fares Data Service';
const description = 'Error page of the Create Fares Data Service';

interface ErrorProps {
    statusCode: number;
}

const Error = ({ statusCode }: ErrorProps): ReactElement => (
    <TwoThirdsLayout title={title} description={description}>
        {statusCode === 404 ? (
            <div>
                <h1 className="govuk-heading-l">Page not found</h1>
                <p className="govuk-body">If you typed the web address, check it is correct.</p>
                <p className="govuk-body">If you pasted the web address, check you copied the entire address.</p>
                <p className="govuk-body">
                    If the web address is correct or you selected a link or button,{' '}
                    <a className="govuk-link" id="feedback-link" href={FEEDBACK_LINK}>
                        contact
                    </a>{' '}
                    us about your fares data.
                </p>
            </div>
        ) : (
            <div>
                <h1 className="govuk-heading-l">Sorry, there is a problem with the service.</h1>
                <p className="govuk-body">Try again later.</p>
                <p className="govuk-body">Your answers have not been saved, use the button below to start again.</p>
                <p className="govuk-body">
                    {' '}
                    <a className="govuk-link" id="feedback-link" href={FEEDBACK_LINK}>
                        Contact
                    </a>{' '}
                    us for assistance.
                </p>
            </div>
        )}

        <br />
        <a href="/" role="button" draggable="false" className="govuk-button" data-module="govuk-button">
            Start again
        </a>
    </TwoThirdsLayout>
);

export const getServerSideProps = (ctx: NextPageContext): {} => {
    return { props: { statusCode: ctx.res?.statusCode } };
};

export default Error;
