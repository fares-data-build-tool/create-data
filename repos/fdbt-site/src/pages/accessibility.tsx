import React, { ReactElement } from 'react';
import { TwoThirdsLayout } from '../layout/Layout';
import DACimage from '../assets/images/DAC_certification.png';

const title = 'Accessibility - Create Fares Data Service';
const description = 'Accessibility page for the Create Fares Data Service';

const Accessibility = (): ReactElement => {
    return (
        <TwoThirdsLayout title={title} description={description}>
            <h1 className="govuk-heading-l">Accessibility statement for the Create Fares Data Service</h1>
            <p className="govuk-body">
                This service is part of the wider GOV.UK website. There’s a separate accessibility statement for the
                main GOV.UK website.
            </p>
            <p className="govuk-body">
                This page only contains information about the Create Fares Data Service, available at&nbsp;
                <a href="https://fares-data.dft.gov.uk">https://fares-data.dft.gov.uk</a>.
            </p>
            <h2 className="govuk-heading-m">Using this service</h2>
            <p className="govuk-body">
                This service is run by the Department for Transport. We want as many people as possible to be able to
                use this service. For example, that means you should be able to:
            </p>
            <ol className="govuk-list govuk-list--bullet">
                <li>change colours, contrast levels and fonts</li>
                <li>zoom in up to 300% without the text spilling off the screen</li>
                <li>get from the start of the service to the end using just a keyboard</li>
                <li>get from the start of the service to the end using speech recognition software</li>
                <li>
                    listen to the service using a screen reader (including the most recent versions of JAWS, NVDA and
                    VoiceOver)
                </li>
            </ol>
            <p className="govuk-body">We’ve also made the text in the service as simple as possible to understand.</p>

            <p className="govuk-body">
                <a href="https://mcmw.abilitynet.org.uk/" rel="external">
                    AbilityNet
                </a>
                &nbsp;has advice on making your device easier to use if you have a disability.
            </p>
            <h2 className="govuk-heading-m">How accessible this service is</h2>
            <p className="govuk-body">
                Some of the language and concepts used throughout this service are aimed at users within the transport
                industry.
            </p>
            <p className="govuk-body">Some documents are in PDF format and are not accessible.</p>
            <h2 className="govuk-heading-m">Feedback and contact information</h2>
            <p className="govuk-body">
                If you have difficulty using this service, <a href="/contact">contact us</a>.
            </p>
            <h2 className="govuk-heading-m">Reporting accessibility problems with this service</h2>
            <p className="govuk-body">
                We’re always looking to improve the accessibility of this service. If you find any problems that are not
                listed on this page or think we’re not meeting accessibility requirements,&nbsp;
                <a href="/contact">contact us</a>.
            </p>
            <h2 className="govuk-heading-m">Enforcement procedure</h2>
            <p className="govuk-body">
                The Equality and Human Rights Commission (EHRC) is responsible for enforcing the Public Sector Bodies
                (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018 (the ‘accessibility
                regulations’). If you’re not happy with how we respond to your complaint,&nbsp;
                <a href="https://www.equalityadvisoryservice.com/">
                    contact the Equality Advisory and Support Service (EASS)
                </a>
                .
            </p>
            <h2 className="govuk-heading-m">Contacting us</h2>
            <p className="govuk-body">
                Find out how to <a href="/contact">contact us</a>.
            </p>
            <h2 className="govuk-heading-m">Technical information about this website’s accessibility</h2>
            <p className="govuk-body">
                The Department for Transport is committed to making its websites accessible, in accordance with the
                Public Sector Bodies (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018.
            </p>
            <h2 className="govuk-heading-m">Compliance status</h2>
            <p className="govuk-body">
                This service is fully compliant with the&nbsp;
                <a href="https://www.w3.org/TR/WCAG21/">Web Content Accessibility Guidelines version 2.1 AA standard</a>
                .
            </p>
            <h2 className="govuk-heading-m"> Preparation of this accessibility statement</h2>
            <p className="govuk-body">
                This statement was prepared on 20 October 2020. It was last reviewed on 20 October 2020.
            </p>
            <p className="govuk-body">
                This website was last tested on 7 October 2020. The test was carried out by the Digital Accessibility
                Centre, certificate found&nbsp;
                <a
                    href="http://digitalaccessibilitycentre.org/index.php/tfn"
                    aria-label="see digital accessibility centre certificate"
                >
                    here
                </a>
                .
            </p>
            <div>
                <img width="150" src={DACimage} alt="DAC AA certification" />
            </div>
        </TwoThirdsLayout>
    );
};

export const getServerSideProps = (): {} => {
    return { props: {} };
};

export default Accessibility;
