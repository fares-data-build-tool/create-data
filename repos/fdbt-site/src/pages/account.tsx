import React, { ReactElement, useState } from 'react';
import { parseCookies } from 'nookies';
import TwoThirdsLayout from '../layout/Layout';
import { ID_TOKEN_COOKIE } from '../constants';
import { getNocFromIdToken, getAttributeFromIdToken, getCsrfToken } from '../utils';
import { ErrorInfo, NextPageContextWithSession } from '../interfaces';
import ErrorSummary from '../components/ErrorSummary';
import BackButton from '../components/BackButton';
import CsrfForm from '../components/CsrfForm';
import FormElementWrapper from '../components/FormElementWrapper';
import { getSessionAttribute } from '../utils/sessions';
import { ACCOUNT_PAGE_ERROR } from '../constants/attributes';
import { getUserAttribute } from '../data/cognito';

const title = 'Account Details - Create Fares Data Service';
const description = 'Account Details page of the Create Fares Data Service';

interface AccountDetailsProps {
    emailAddress: string;
    nocCode: string;
    csrfToken: string;
    multiOperatorEmailPreference: boolean;
    errors: ErrorInfo[];
}

const AccountDetails = ({
    emailAddress,
    nocCode,
    csrfToken,
    multiOperatorEmailPreference,
    errors,
}: AccountDetailsProps): ReactElement => {
    const [multiOpEmailPreference, setMultiOpEmailPreference] = useState(multiOperatorEmailPreference);
    const passwordDots: ReactElement[] = [];
    for (let i = 0; i < 8; i += 1) {
        passwordDots.push(<span key={i} className="dot" />);
    }

    return (
        <TwoThirdsLayout title={title} description={description}>
            <CsrfForm action="/api/updateEmailPreference" method="post" csrfToken={csrfToken}>
                <ErrorSummary errors={errors} />
                <div className="account-details-page">
                    <BackButton href="/home" />
                    <h1 className="govuk-heading-xl">Account Details</h1>
                    <div className="content-wrapper">
                        <p className="govuk-body govuk-!-font-weight-bold content-one-quarter">Email Address</p>
                        <p className="govuk-body email-content content-three-quarters">{emailAddress}</p>
                    </div>
                    <div className="content-wrapper">
                        <p className="govuk-body govuk-!-font-weight-bold content-one-quarter">Password</p>
                        <span className="password-dots">{passwordDots}</span>
                        <span className="change-password">
                            <a
                                href="/changePassword"
                                role="button"
                                draggable="false"
                                className="govuk-button"
                                data-module="govuk-button"
                                id="change-password-button"
                                aria-label="change password"
                            >
                                Change
                            </a>
                        </span>
                    </div>
                    <div className="content-wrapper govuk-!-margin-bottom-8">
                        <p className="govuk-body govuk-!-font-weight-bold content-one-quarter">Operator</p>
                        <p className="govuk-body content-three-quarters">{nocCode.replace(/\|/g, ', ')}</p>
                    </div>

                    <fieldset className="govuk-fieldset">
                        <legend className="govuk-fieldset__legend govuk-fieldset__legend--l">
                            <p className="govuk-heading-m">Email notifications</p>
                            <p className="govuk-heading-s">Multi-operator fare alerts</p>
                        </legend>
                        <FormElementWrapper
                            errors={errors}
                            errorId="radio-multi-op-email-pref"
                            errorClass="govuk-radios--error"
                        >
                            <div className="govuk-radios govuk-radios--inline ml-auto" data-module="govuk-radios">
                                <div className="govuk-radios__item">
                                    <input
                                        className="govuk-radios__input"
                                        id="multiOpEmailPref-on"
                                        name="multiOpEmailPref"
                                        checked={multiOpEmailPreference}
                                        onChange={() => setMultiOpEmailPreference(true)}
                                        type="radio"
                                        value="true"
                                    />
                                    <label className="govuk-label govuk-radios__label" htmlFor="multiOpEmailPref-on">
                                        On
                                    </label>
                                </div>
                                <div className="govuk-radios__item">
                                    <input
                                        className="govuk-radios__input"
                                        id="multiOpEmailPref-off"
                                        name="multiOpEmailPref"
                                        checked={!multiOpEmailPreference}
                                        onChange={() => setMultiOpEmailPreference(false)}
                                        type="radio"
                                        value="false"
                                    />
                                    <label className="govuk-label govuk-radios__label" htmlFor="multiOpEmailPref-off">
                                        Off
                                    </label>
                                </div>
                            </div>
                        </FormElementWrapper>
                    </fieldset>
                </div>
                <input
                    type="submit"
                    name="saveChanges"
                    value="Save changes"
                    id="save-changes-button"
                    data-module="govuk-button"
                    className="govuk-button govuk-!-margin-right-4"
                />
                <a
                    href="/home"
                    role="button"
                    draggable="false"
                    className="govuk-button govuk-button--secondary"
                    data-module="govuk-button"
                    id="cancel-button"
                >
                    Cancel changes
                </a>
            </CsrfForm>
        </TwoThirdsLayout>
    );
};

export const getServerSideProps = async (ctx: NextPageContextWithSession): Promise<{ props: AccountDetailsProps }> => {
    const cookies = parseCookies(ctx);
    if (!cookies[ID_TOKEN_COOKIE]) {
        throw new Error('Necessary attributes not found to show account details');
    }
    const noc = getNocFromIdToken(ctx);
    const email = getAttributeFromIdToken(ctx, 'email');

    if (!email || !noc) {
        throw new Error('Could not extract the user email address and/or noc code from their ID token');
    }

    const host = ctx.req.headers.host;
    const multiOperatorEmailPreference =
        host && host.startsWith('localhost') ? 'false' : await getUserAttribute(email, 'custom:multiOpEmailEnabled');

    const errors = getSessionAttribute(ctx.req, ACCOUNT_PAGE_ERROR);

    return {
        props: {
            emailAddress: email,
            nocCode: noc,
            csrfToken: getCsrfToken(ctx),
            multiOperatorEmailPreference: multiOperatorEmailPreference === 'true' ?? false,
            errors: errors ?? [],
        },
    };
};

export default AccountDetails;
