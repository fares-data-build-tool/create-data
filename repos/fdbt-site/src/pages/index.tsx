import '../design/Pages.scss';
import React, { ReactElement } from 'react';
import { NextPage } from 'next';
import Layout from '../layout/Layout';

const title = 'Fares data build tool';
const description = 'Fares data build tool is a service that allows you to generate data in NeTEx format';

const Home: NextPage = (): ReactElement => (
    <Layout title={title} description={description}>
        <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
            <h1 className="govuk-heading-xl">Fares data build tool</h1>
            <p className="govuk-body-l">Use this service to:</p>
            <div className="govuk-body-l">
                <ul className="index-page-list">
                    <li>Add fares to a newly registered bus service</li>
                    <li>Update fares on an existing bus service</li>
                    <li>Generate fares data for services in NeTEx format</li>
                </ul>
            </div>

            <a
                href="/operator"
                role="button"
                draggable="false"
                className="govuk-button govuk-button--start"
                data-module="govuk-button"
                id="start-now-button"
            >
                Start now
                <svg
                    className="govuk-button__start-icon"
                    xmlns="http://www.w3.org/2000/svg"
                    width="17.5"
                    height="19"
                    viewBox="0 0 33 40"
                    role="presentation"
                    focusable="false"
                >
                    <path fill="currentColor" d="M0 0h13l20 20-20 20H0l20-20z" />
                </svg>
            </a>
        </main>
    </Layout>
);

Home.getInitialProps = (): {} => {
    return {};
};

export default Home;
