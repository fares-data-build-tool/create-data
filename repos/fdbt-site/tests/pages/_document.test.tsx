/* eslint-disable react/jsx-props-no-spreading */
import * as React from 'react';
import { shallow } from 'enzyme';
import { DocumentProps } from 'next/dist/pages/_document';
import MyDocument from '../../src/pages/_document';

describe('_document', () => {
    const props: DocumentProps = {
        html: '',
        __NEXT_DATA__: {
            props: {},
            page: '',
            query: {
                '': '',
            },
            buildId: '',
        },
        dangerousAsPath: '',
        ampPath: '',
        inAmpMode: false,
        hybridAmp: false,
        isDevelopment: false,
        dynamicImports: [],
        canonicalBase: '',
        headTags: [{}],
        buildManifest: {
            ampDevFiles: [''],
            ampFirstPages: [''],
            devFiles: [''],
            lowPriorityFiles: [''],
            pages: {
                '/_app': [''],
            },
            polyfillFiles: [''],
        },
        devOnlyCacheBusterQueryString: '',
        docComponentsRendered: {},
        scriptLoader: {},
    };

    it('should render correctly', () => {
        const tree = shallow(
            <MyDocument {...props} nonce="" isAuthed csrfToken="" url="" showCookieBanner allowTracking />,
        );

        expect(tree).toMatchSnapshot();
    });

    it('should not show the cookie banner when the showCookieBanner attribute is false', () => {
        const tree = shallow(
            <MyDocument {...props} nonce="" isAuthed csrfToken="" url="" showCookieBanner={false} allowTracking />,
        );

        expect(tree).toMatchSnapshot();
    });
});
