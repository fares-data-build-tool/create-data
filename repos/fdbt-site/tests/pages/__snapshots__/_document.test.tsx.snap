// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_document should not show the cookie banner when the showCookieBanner attribute is false 1`] = `
<Html
  className="govuk-template app-html-class flexbox no-flexboxtweener"
  lang="en"
>
  <Head
    nonce=""
  >
    <script
      async={true}
      nonce=""
      src="https://www.googletagmanager.com/gtag/js?id=UA-173062045-1"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window['ga-disable-UA-173062045-1'] = false;
                                        window.dataLayer = window.dataLayer || [];
                                        function gtag(){dataLayer.push(arguments);}
                                        gtag('js', new Date());
                                        gtag('config', 'UA-173062045-1', {'anonymize_ip': true});",
        }
      }
      nonce=""
    />
  </Head>
  <body
    className="govuk-template__body app-body-class"
  >
    <a
      className="govuk-skip-link"
      href="#main-content"
    >
      Skip to main content
    </a>
    <div
      id="js-cookie-banner"
    />
    <Header
      csrfToken=""
      isAuthed={true}
    />
    <Main />
    <NextScript
      nonce=""
    />
    <script
      nonce=""
      src="/scripts/all.js"
    />
    <script
      nonce=""
    >
      window.GOVUKFrontend.initAll()
    </script>
  </body>
</Html>
`;

exports[`_document should render correctly 1`] = `
<Html
  className="govuk-template app-html-class flexbox no-flexboxtweener"
  lang="en"
>
  <Head
    nonce=""
  >
    <script
      async={true}
      nonce=""
      src="https://www.googletagmanager.com/gtag/js?id=UA-173062045-1"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window['ga-disable-UA-173062045-1'] = false;
                                        window.dataLayer = window.dataLayer || [];
                                        function gtag(){dataLayer.push(arguments);}
                                        gtag('js', new Date());
                                        gtag('config', 'UA-173062045-1', {'anonymize_ip': true});",
        }
      }
      nonce=""
    />
  </Head>
  <body
    className="govuk-template__body app-body-class"
  >
    <a
      className="govuk-skip-link"
      href="#main-content"
    >
      Skip to main content
    </a>
    <div
      id="js-cookie-banner"
    />
    <div
      className="no-js-cookie-banner"
    >
      <Component />
    </div>
    <Header
      csrfToken=""
      isAuthed={true}
    />
    <Main />
    <NextScript
      nonce=""
    />
    <script
      nonce=""
      src="/scripts/all.js"
    />
    <script
      nonce=""
    >
      window.GOVUKFrontend.initAll()
    </script>
  </body>
</Html>
`;
